src/Sylius/Bundle/OrderBundle/Form/DataMapper/OrderItemQuantityDataMapper.php

```diff
@@ -60,8 +60,7 @@ public function mapFormsToData($forms, &$data)
 
         $formsOtherThanQuantity = [];
         foreach ($forms as $key => $form) {
-            if ('quantity' === $form->getName()) {
-                $targetQuantity = $form->getData();
+            if ('quantity' === $form->getName() && $targetQuantity = $form->getData()) {
```

Comment:
I'm going to have a look at the first error, maybe that helps to figure out this second one. 
I don't know why `$form->getData()` is returning `null` instead of the variant.